docs/generator.go

```diff
@@ -2,42 +2,86 @@ package main
 
 import (
 	"encoding/json"
+	"flag"
 	"fmt"
 	"io/ioutil"
 	"log"
 	"os"
+	"strings"
 	"text/template"
+
+	"github.com/tidwall/gjson"
 )
 
 func main() {
-	contents, err := ioutil.ReadFile("../schema.json")
+	var conf, doc, tmpl string
+	flag.StringVar(&conf, "conf", "schema.json", "Schema file")
+	flag.StringVar(&doc, "doc", "docs/doc.md", "Markdown file")
+	flag.StringVar(&tmpl, "tmpl", "docs/user/test.md", "Output file")
+	content, err := ioutil.ReadFile(conf)
 	if err != nil {
 		log.Fatal(err)
 	}
+	f, err := os.OpenFile(doc, os.O_APPEND, 0666)
+	value := gjson.GetBytes(content, "properties")
+	value.ForEach(func(key, value gjson.Result) bool {
+		_, err = f.WriteString("\n# " + strings.Title(key.String()) + "\n" + "This section is of " + "{{.Properties." + strings.Title(key.String()) + ".Title.Description}} type that tells about the basic information of the user and consists of the following sub-sections:\n") //print name of sections
+		if err != nil {
+			log.Fatal(err)
+		}
+		println(key.String())
+		result := gjson.GetBytes(content, "properties."+key.String())
+		result.ForEach(func(key1, value gjson.Result) bool {
+			_, err = f.WriteString("##    {{" + key1.String() + "}}" + "\n") //print name of sub-sections
+			if err != nil {
+				log.Fatal(err)
+			}
+			print("  ")
+			println(key1.String())
 
+			if key1.String() == "properties" {
+				result1 := gjson.GetBytes(content, "properties."+key.String()+"."+key1.String())
+				result1.ForEach(func(key2, value gjson.Result) bool {
+					_, err := f.WriteString("###        {{" + key2.String() + "}}" + "\nSub-section of type {{.Properties." + strings.Title(key.String()) + ".Properties." + strings.Title(key2.String()) + ".Type}},used to specify the " + key2.String() + " of the person\nThe schema snippet can be shown below:\n        " + key2.String() + ":\n        " + value.String() + "\n") //print properties of subsections
+					if err != nil {
+						log.Fatal(err)
+					}
+					print("         ")
+					println(key2.String())
+					return true
+				})
+			}
+			if key1.String() == "items" {
+				result1 := gjson.GetBytes(content, "properties."+key.String()+".items.properties")
+				result1.ForEach(func(key2, value gjson.Result) bool {
+					_, err := f.WriteString("###        {{" + key2.String() + "}}" + "\nSub-section of type {{.Properties." + strings.Title(key.String()) + ".Items.Properties." + strings.Title(key2.String()) + ".Type}},used to specify the " + key2.String() + " of the person\nThe schema snippet can be shown below:\n        " + key2.String() + ":\n        " + value.String() + "\n") //print items of subsections
```

Comment:
Any reason for the excessive whitespace?

Reply:
@stp-ip Ah sorry, by mistake will correct it on next commit